thunder/dynamo/report.py

```diff
@@ -490,7 +506,8 @@ def write_repro(
 
             code_str += textwrap.indent(check_str, "    ")
 
-        code_str = f"{code_str}\ntest_{self.graph_name}()"
+        code_str = f"{code_str}\n{main_code.format(graph_name=self.graph_name)}\n{comment_str}"
+        code_str = black.format_str(code_str, mode=format_mode)
```

Comment:
Instead of unconditionally formatting the string here (and importing `black` at the start of this file), instead query if `black` is available following this pattern

https://github.com/Lightning-AI/lightning-thunder/blob/b4640fef634522f3d4730457dd09027cc759be91/thunder/executors/cudnn_layernormex.py#L10

and only import `black` and format the code using it if it's available. In the future different formatters could be supported, too